src/fparser/two/tests/fortran2003/test_primary_r701.py

```diff
@@ -0,0 +1,74 @@
+# Copyright (c) 2019 Science and Technology Facilities Council
+
+# All rights reserved.
+
+# Modifications made as part of the fparser project are distributed
+# under the following license:
+
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are
+# met:
+
+# 1. Redistributions of source code must retain the above copyright
+# notice, this list of conditions and the following disclaimer.
+
+# 2. Redistributions in binary form must reproduce the above copyright
+# notice, this list of conditions and the following disclaimer in the
+# documentation and/or other materials provided with the distribution.
+
+# 3. Neither the name of the copyright holder nor the names of its
+# contributors may be used to endorse or promote products derived from
+# this software without specific prior written permission.
+
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+'''Test Fortran 2003 rule R701 : primary type.
+
+'''
+import pytest
+
+import fparser.two.Fortran2003 as f2003
+import fparser.two.utils
+
+
+# A list of [Expression, Expected Type, ToString] for the test cases.
+cases = [
+    ['1.2e-03', f2003.Real_Literal_Constant, '1.2E-03'],
+    ['(1, n)', f2003.Complex_Literal_Constant, '(1, n)'],
+    ['.true.', f2003.Logical_Literal_Constant, '.TRUE.'],
+    ['"hey a()c"', f2003.Char_Literal_Constant, '"hey a()c"'],
+    ['b"0101"', f2003.Binary_Constant, 'B"0101"'],
+
```

Comment:
Could you please add a comment explaining which test corresponds to which subclass as it is not clear.

If there are any subclass tests missing for valid reasons (i.e. they are not possible to test) then please add comments explaining why, create an issue and reference the issue in the comment.